File: max_element_in_array_recursive.py
def find_max(array):

    # base case
    if len(array) == 1:
        return array[0]
    
    # recursive case
    array[-2] = (array[-1] if array[-1]>array[-2] else array[-2])
    del array[-1]
    return find_max(array)
    # [3,5]

print(find_max([10,3,45,1,6,32]))
# [10,3,45,1,6,32]

File: max_element_in_array_recursive.js
const findMax_recursion = array => {

    // base case
    if (array.length == 1){
        return array[0];
    }
    // recursive case
    if ( array[array.length-1] > array[array.length-2] ){
        array[array.length-2] = array[array.length-1];
    }
    array.pop();
    return findMax_recursion(array);
}

console.log(findMax_recursion([3,10,-3,12,5,12,-61,2313,2]));